pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -854,6 +859,11 @@ public class Utils {
 
             // Now add all the other job properties we know need to be added.
             jobPropertiesToApply.each { p ->
+                // Remove the existing instance(s) of the property class before we add the new one. We're looping and
+                // removing multiple to deal with the results of JENKINS-44809.
+                while (j.getProperty(p.class) != null) {
+                    j.removeProperty(p.class)
```

Comment:
...and they should probably remove all instances with the given `Descriptor.id` rather than just the first...